packages/extension-ui/src/components/SplashHandler.tsx

```diff
@@ -43,11 +43,7 @@ function SplashHandler({
   };
 
   useEffect(() => {
-    if (!isContentVisible) {
-      return;
-    }
```

Comment:
Don't we want to start the timeout only after the animation started? Btw maybe renaming `isContentVisible` to `hasAnimationStarted` (and **maybe** later do `const isContentVisible = hasAnimationStarted` to use `isContentVisible` in the render return for semantics) would be more readable and help reckoning about the flow. At least I found myself using more levels of brain stack trace to figure out what's going on, without it being immediately clear that this `useEffect` was short-circuited on the rule of _the video already running_, not _content being visible_.

Reply:
I happened to me in storybook (but I assume it's theoretically possible anywhere) that the video didn't start - some error was thrown and `onStarted` callback was never called. I wanted to make sure, that even if this happens users will be able to use Signer

Reply:
after removing the condition from `useEffect` the `isContentVisible` (I'll change it to `isContentRendered`) make more sense to me -- in onStarted we set content to be rendered. In case `onStarted` isn't called after 2s we set it to render anyway, and we render children base on `isContentRendered`